components/earnings/commons/earnings_base.lua

```diff
@@ -0,0 +1,185 @@
+
+	if String.isEmpty(player) then
+		return 0
+	end
+	if not Logic.readBool(args.noRedirect) then
+		player = mw.ext.TeamLiquidIntegration.resolve_redirect(player)
+	end
+
+	local prefix = args.prefix or 'p'
+
+	local playerNumber = tonumber(args.playerNumber) or 10
+	if playerNumber <=0 then
+		error('"playerNumber" has to be >= 1')
+	end
+
+	local playerConditions = '([[participant::' .. player .. ']]'
+	for playerIndex = 1, playerNumber do
+		playerConditions = playerConditions .. ' OR [[players_' .. prefix .. playerIndex .. '::' .. player .. ']]'
+	end
+	playerConditions = playerConditions .. ')'
+
+	return Earnings._calculate(playerConditions, args.year, args.mode, args.startYear, Earnings.divisionFactor)
+end
+
+---
+-- Entry point for teams
+-- @team - the team for which the earnings shall be calculated
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+-- @noRedirect - (optional) player redirects get not resolved before query
+-- @startYear - (optional) query yearly earning starting with that year and return the values in a lua table
+function Earnings.calculateForTeam(args)
+	args = args or {}
+	local team = args.team
+
+	if String.isEmpty(team) then
+		return 0
+	end
+	if not Logic.readBool(args.noRedirect) then
+		team = mw.ext.TeamLiquidIntegration.resolve_redirect(team)
+	end
+
+	local teamConditions = '([[participant::' .. team .. ']] OR [[extradata_participantteam::' .. team .. ']])'
+
+	return Earnings._calculate(teamConditions, args.year, args.mode, args.startYear, Earnings._divisionFactorOne)
+end
+
+---
+-- Calculates earnings for this participant in a certain mode
+-- @participantCondition - the condition to find the player/team
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+function Earnings._calculate(conditions, year, mode, startYear, divisionFactor)
+	conditions = Earnings._buildConditions(conditions, year, mode, startYear)
+
+	if String.isNotEmpty(startYear) then
+		return Earnings._calculatePerYear(conditions, divisionFactor)
+	end
+
+	local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = conditions,
+		query = 'sum::prizemoney, mode',
+		groupby = 'mode asc'
+	})
+
+	local totalEarnings = 0
+
+	for _, item in ipairs(lpdbQueryData) do
+		if item['sum_prizemoney'] ~= nil then
+			local prizeMoney = item['sum_prizemoney']
+			totalEarnings = totalEarnings + (prizeMoney / divisionFactor(item['mode']))
+		end
+	end
+
+	return MathUtils._round(totalEarnings, 2)
+end
+
+function Earnings._calculatePerYear(conditions, divisionFactor)
+	local totalEarningsOfYear = {}
+	local totalEarnings = {}
+	totalEarnings.total = 0
+
+	local offset = 0
+	local count = 5000
+	while count == 5000 do
+		local lpdbQueryData = mw.ext.LiquipediaDB.lpdb('placement', {
+			conditions = conditions,
+			query = 'prizemoney, mode, date',
+			limit = 5000,
+			offset = offset
+		})
```

Comment:
Thought, does `groupby = 'date_year'` work?

Reply:
don't think so, hence implemented it this way, will test later today

Reply:
nope does not work (like with stuff in extradata can't group by that)